src/core/mac/mac.cpp

```diff
@@ -933,6 +944,7 @@ void Mac::TransmitDoneTask(otRadioFrame *aFrame, bool aRxPending, otError aError
     case kStateTransmitData:
         if (aRxPending)
         {
+            mWaitingForData = true;
```

Comment:
We should only execute this block this if the transmit frame is a IEEE 802.15.4 Data Request command frame.